test/supavisor/db_handler_test.exs

```diff
@@ -166,6 +166,84 @@ defmodule Supavisor.DbHandlerTest do
     end
   end
 
+  describe "handle_event/4 info tcp authentication authentication_sasl_password payload events" do
+    setup do
+      {a, b} = sockpair()
+
+      data = %{
+        auth: %{
+          user: "user",
+          require_user: false,
+          secrets: fn -> %{user: "user", password: "pass"} end
+        },
+        sock: {:gen_tcp, a},
+        nonce: "some nonce"
+      }
+
+      %{data: data, b: b}
+    end
+
+    test "handles SASL authentication and sets nonce", %{data: data, b: b} do
+      # `82` is `?R`, which identifies the payload tag as `:authentication`
+      # `0, 0, 0, 22` is the packet length
+      # `0, 0, 0, 10` is the authentication type, identified as `:authentication_sasl_password`
+      payload = "SCRAM-SHA-256" <> <<0>>
+      bin = <<82, 0, 0, 0, 22, 0, 0, 0, 10, payload::binary>>
```

Comment:
```suggestion
      bin = <<?R, 22::32  10::32, "SCRAM-SHA-256", 0>>
```

Reply:
Yeap. I was following the previous tests but I really prefer like that.